diff -urN demo408/app/assets/stylesheets/application.css demo414/app/assets/stylesheets/application.css
--- demo408/app/assets/stylesheets/application.css	2014-07-17 23:34:23.979557689 +0300
+++ demo414/app/assets/stylesheets/application.css	2014-07-17 23:53:30.617827215 +0300
@@ -5,9 +5,11 @@
  * Any CSS and SCSS file within this directory, lib/assets/stylesheets, vendor/assets/stylesheets,
  * or vendor/assets/stylesheets of plugins, if any, can be referenced here using a relative path.
  *
- * You're free to add application-wide styles to this file and they'll appear at the top of the
- * compiled file, but it's generally better to create a new file per style scope.
+ * You're free to add application-wide styles to this file and they'll appear at the bottom of the
+ * compiled file so the styles you add here take precedence over styles defined in any styles
+ * defined in the other CSS/SCSS files in this directory. It is generally better to create a new
+ * file per style scope.
  *
- *= require_self
  *= require_tree .
+ *= require_self
  */
diff -urN demo408/app/views/layouts/application.html.erb demo414/app/views/layouts/application.html.erb
--- demo408/app/views/layouts/application.html.erb	2014-07-17 23:34:23.985557465 +0300
+++ demo414/app/views/layouts/application.html.erb	2014-07-17 23:53:30.623826992 +0300
@@ -1,9 +1,9 @@
 <!DOCTYPE html>
 <html>
 <head>
-  <title>Demo408</title>
-  <%= stylesheet_link_tag    "application", media: "all", "data-turbolinks-track" => true %>
-  <%= javascript_include_tag "application", "data-turbolinks-track" => true %>
+  <title>Demo414</title>
+  <%= stylesheet_link_tag    'application', media: 'all', 'data-turbolinks-track' => true %>
+  <%= javascript_include_tag 'application', 'data-turbolinks-track' => true %>
   <%= csrf_meta_tags %>
 </head>
 <body>
diff -urN demo408/bin/rails demo414/bin/rails
diff -urN demo408/bin/rake demo414/bin/rake
diff -urN demo408/bin/spring demo414/bin/spring
diff -urN demo408/config/application.rb demo414/config/application.rb
diff -urN demo408/config/database.yml demo414/config/database.yml
--- demo408/config/database.yml	2014-07-17 23:34:24.102553106 +0300
+++ demo414/config/database.yml	2014-07-17 23:53:30.669825277 +0300
@@ -3,23 +3,23 @@
 #
 #   Ensure the SQLite 3 gem is defined in your Gemfile
 #   gem 'sqlite3'
-development:
+#
+default: &default
   adapter: sqlite3
-  database: db/development.sqlite3
   pool: 5
   timeout: 5000
 
+development:
+  <<: *default
+  database: db/development.sqlite3
+
 # Warning: The database defined as "test" will be erased and
 # re-generated from your development database when you run "rake".
 # Do not set this db to the same as development or production.
 test:
-  adapter: sqlite3
+  <<: *default
   database: db/test.sqlite3
-  pool: 5
-  timeout: 5000
 
 production:
-  adapter: sqlite3
+  <<: *default
   database: db/production.sqlite3
-  pool: 5
-  timeout: 5000
diff -urN demo408/config/environments/production.rb demo414/config/environments/production.rb
--- demo408/config/environments/production.rb	2014-07-17 23:34:24.018556236 +0300
+++ demo414/config/environments/production.rb	2014-07-17 23:53:30.644826209 +0300
@@ -1,11 +1,11 @@
-Demo408::Application.configure do
+Rails.application.configure do
   # Settings specified here will take precedence over those in config/application.rb.
 
   # Code is not reloaded between requests.
   config.cache_classes = true
 
   # Eager load code on boot. This eager loads most of Rails and
-  # your application in memory, allowing both thread web servers
+  # your application in memory, allowing both threaded web servers
   # and those relying on copy on write to perform better.
   # Rake tasks automatically ignore this option for performance.
   config.eager_load = true
@@ -32,8 +32,7 @@
   # Generate digests for assets URLs.
   config.assets.digest = true
 
-  # Version of your assets, change this if you want to expire all your assets.
-  config.assets.version = '1.0'
+  # `config.assets.precompile` has moved to config/initializers/assets.rb
 
   # Specifies the header that your server uses for sending files.
   # config.action_dispatch.x_sendfile_header = "X-Sendfile" # for apache
@@ -66,7 +65,7 @@
   # config.action_mailer.raise_delivery_errors = false
 
   # Enable locale fallbacks for I18n (makes lookups for any locale fall back to
-  # the I18n.default_locale when a translation can not be found).
+  # the I18n.default_locale when a translation cannot be found).
   config.i18n.fallbacks = true
 
   # Send deprecation notices to registered listeners.
@@ -77,4 +76,7 @@
 
   # Use default logging formatter so that PID and timestamp are not suppressed.
   config.log_formatter = ::Logger::Formatter.new
+
+  # Do not dump schema after migrations.
+  config.active_record.dump_schema_after_migration = false
 end
diff -urN demo408/config/initializers/assets.rb demo414/config/initializers/assets.rb
--- demo408/config/initializers/assets.rb	1970-01-01 03:00:00.000000000 +0300
+++ demo414/config/initializers/assets.rb	2014-07-17 23:53:30.651825948 +0300
@@ -0,0 +1,8 @@
+# Be sure to restart your server when you modify this file.
+
+# Version of your assets, change this if you want to expire all your assets.
+Rails.application.config.assets.version = '1.0'
+
+# Precompile additional assets.
+# application.js, application.css, and all non-JS/CSS in app/assets folder are already added.
+# Rails.application.config.assets.precompile += %w( search.js )
diff -urN demo408/config/initializers/secret_token.rb demo414/config/initializers/secret_token.rb
--- demo408/config/initializers/secret_token.rb	2014-07-17 23:34:24.072554224 +0300
+++ demo414/config/initializers/secret_token.rb	1970-01-01 03:00:00.000000000 +0300
@@ -1,12 +0,0 @@
-# Be sure to restart your server when you modify this file.
-
-# Your secret key is used for verifying the integrity of signed cookies.
-# If you change this key, all old signed cookies will become invalid!
-
-# Make sure the secret is at least 30 characters and all random,
-# no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
-
-# Make sure your secret_key_base is kept private
-# if you're sharing your code publicly.
-Demo408::Application.config.secret_key_base = 'c02d636e80c91830cd86928cd90734e1715daf05c631b0b64c9a63f095d613a72c92d4b7e50362f4c0dc20bc73b4ee17cf56f7e25b603fbee669f4a6af48cac6'
diff -urN demo408/config/secrets.yml demo414/config/secrets.yml
--- demo408/config/secrets.yml	1970-01-01 03:00:00.000000000 +0300
+++ demo414/config/secrets.yml	2014-07-17 23:53:30.638826433 +0300
@@ -0,0 +1,22 @@
+# Be sure to restart your server when you modify this file.
+
+# Your secret key is used for verifying the integrity of signed cookies.
+# If you change this key, all old signed cookies will become invalid!
+
+# Make sure the secret is at least 30 characters and all random,
+# no regular words or you'll be exposed to dictionary attacks.
+# You can use `rake secret` to generate a secure secret key.
+
+# Make sure the secrets in this file are kept private
+# if you're sharing your code publicly.
+
+development:
+  secret_key_base: 1a8fea9fc560c3ae23d5fcf3ab74b4dea219e5a6b7a7360c669651dc8c2d665f939d4a0ebfcb70df4c86f99fa08dd9b68cbc71ad16bc40981f3bf3971dd97848
+
+test:
+  secret_key_base: a854d24e9b659d6191e99a7629fabd4421da62ba212fe36ec654cf2a2878e2fe2ddd726f29fce488bc0dc4034185d51c241d71882acb68225758213609731e84
+
+# Do not keep production secrets in the repository,
+# instead read values from the environment.
+production:
+  secret_key_base: <%= ENV["SECRET_KEY_BASE"] %>
diff -urN demo408/public/404.html demo414/public/404.html
diff -urN demo408/public/422.html demo414/public/422.html
diff -urN demo408/public/500.html demo414/public/500.html
diff -urN demo408/Rakefile demo414/Rakefile
diff -urN demo408/test/test_helper.rb demo414/test/test_helper.rb
