diff -ur pmount-0.9.17.orig/src/fs.c pmount-0.9.17/src/fs.c
--- pmount-0.9.17.orig/src/fs.c	2007-12-14 20:44:34 +0200
+++ pmount-0.9.17/src/fs.c	2008-01-08 15:44:17 +0200
@@ -19,10 +19,10 @@
  * to command line options. It is anyway better not to use it...
  */
 static struct FS supported_fs[] = {
-    { "udf", "nosuid,nodev,user", 1, "000", ",iocharset=%s" },
-    { "iso9660", "nosuid,nodev,user", 1, NULL, ",iocharset=%s" },
     { "vfat", "nosuid,nodev,user,quiet,shortname=mixed", 1, "077", 
       ",iocharset=%s",",fmask=%04o,dmask=%04o"},
+    { "udf", "nosuid,nodev,user", 1, "000", ",iocharset=%s" },
+    { "iso9660", "nosuid,nodev,user", 1, NULL, ",iocharset=%s" },
     { "hfsplus", "nosuid,nodev,user", 1, NULL, 0 },
     { "hfs", "nosuid,nodev,user", 1, "077", NULL, 
       ",file_umask=%04o,dir_umask=%04o"},
