diff -urN pmount-0.9.19.orig/src/fs.c pmount-0.9.19/src/fs.c
--- pmount-0.9.19.orig/src/fs.c	2009-08-01 23:36:52 +0300
+++ pmount-0.9.19/src/fs.c	2009-08-01 23:40:31 +0300
@@ -26,6 +26,7 @@
     { "hfsplus", "nosuid,nodev,user", 1, NULL, 0 },
     { "hfs", "nosuid,nodev,user", 1, "077", NULL, 
       ",file_umask=%04o,dir_umask=%04o"},
+    { "ext4", "nodev,noauto,nosuid,user,errors=continue", 0, NULL, 0 },
     { "ext3", "nodev,noauto,nosuid,user,errors=continue", 0, NULL, 0 },
     { "ext2", "nodev,noauto,nosuid,user,errors=continue", 0, NULL, 0 },
     { "reiserfs", "nodev,noauto,nosuid,user", 0, NULL, 0 },
